lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -14,19 +14,35 @@
 
 
 class LindiH5pyFile(h5py.File):
-    def __init__(self, _file_object: Union[h5py.File, zarr.Group]):
+    def __init__(self, _file_object: Union[h5py.File, zarr.Group], *, _zarr_store: Union[ZarrStore, None] = None, _mode: Literal["r", "r+"] = "r"):
         """
         Do not use this constructor directly. Instead, use:
         from_reference_file_system, from_zarr_store, from_zarr_group,
         or from_h5py_file
         """
         self._file_object = _file_object
+        self._zarr_store = _zarr_store
+        self._mode: Literal['r', 'r+'] = _mode
         self._the_group = LindiH5pyGroup(_file_object, self)
 
     @staticmethod
-    def from_reference_file_system(rfs: Union[dict, str]):
+    def from_reference_file_system(rfs: Union[dict, str], mode: Literal["r", "r+"] = "r"):
         """
         Create a LindiH5pyFile from a reference file system.
+
+        Parameters
+        ----------
+        rfs : Union[dict, str]
+            The reference file system. This can be a dictionary or a URL or path
+            to a .zarr.json file.
+        mode : Literal["r", "r+"], optional
+            The mode to open the file object in (default: "r"). If the mode is
+            "r", the file object will be read-only. If the mode is "r+", the
+            file will be read-write. However, if the rfs is a string (URL or
+            path), the file itself will not be modified on changes, but the
+            internal in-memory representation will be modified. Use
+            to_reference_file_system() to export the updated reference file
+            system.
```

Comment:
```suggestion
            system to the same file or a new file.
```

Reply:
> the file itself will not be modified on changes

This does not seem intuitive. We can merge but I'll think about what might be a more intuitive interface for users.

Reply:
I agree, we'll need to think about it.